src/Segment.java

```diff
@@ -0,0 +1,100 @@
+import java.lang.reflect.Array;
+import java.sql.Date;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+
+public class Segment extends Point {
+
+    private  String lat_destination , lng_destination , id_ride_destination, timestamp_destination;
+    private double U, distance, delta_time_hours;
+    private  int new_timestamp = Integer.parseInt(timestamp);
+    private  float new_id_ride = Float.parseFloat(id_ride);
+    private  double new_lng = Double.parseDouble(lng);
+    private double new_lat = Double.parseDouble(lat);
+
+    public Segment(String lat,
+                   String lng,
+                   String id_ride,
+                   String timestamp,
+                   String lat_destination,
+                   String lng_destination,
+                   String id_ride_destination,
+                   String timestamp_destination) {
+        super(lat, lng, id_ride, timestamp);
+
+        this.lat_destination = lat_destination;
+        this.lng_destination = lng_destination;
+        this.id_ride_destination = id_ride_destination;
+        this.timestamp_destination = timestamp_destination;
+
+    }
+
+    public double calculate_speed() {
```

Comment:
You should split this method in 2, one for calculate distance and one more that calculate speed = distance/time. In that way you can reuse distance calculation without calculate the speed.